pkg/operator/ceph/object/health.go

```diff
@@ -170,6 +172,11 @@ func (c *bucketChecker) checkObjectStoreHealth() error {
 	if err != nil {
```

Comment:
what about using NewTestOnlyS3Agent() instead NewS3Agent, I am bit unclear do NewS3Agent() itself will fail while setting the connection or following up s3apis fail? Secondly, if we are an insecure connection here and all other parts uses a secure connection all other operations like `bucket provisioning` or `cephobjectstoreuser` creation due to certificate issue but healthchecker passes. Does that provide wrong impression for the users?

Reply:
> what about using NewTestOnlyS3Agent() instead NewS3Agent, I am bit unclear do NewS3Agent() itself will fail while setting the connection or only following up s3apis fail? 

It's a valid point I guess.

> Secondly, if we are an insecure connection here and all other parts uses a secure connection for operations like `bucket provisioning` or `cephobjectstoreuser` creation due to certificate issue but `healthchecker` passes. Does that provide wrong impression for the users?

In this error case, the users are not impacted because the request is the correct DNS name.